benchmarks/src/jmh/kotlin/benchmarks/ChannelBenchmark.kt

```diff
@@ -0,0 +1,151 @@
+
+        lateinit var channel: Channel<Int>
+
+        val maxCount = 100000000
+        val list = ArrayList<Int>(maxCount).apply {
+            repeat(maxCount) { add(it) }
+        }
+
+        @Setup(Level.Invocation)
+        fun createPrefilledChannel() {
+            channel = Channel(Channel.UNLIMITED)
+            repeat(prefill) {
+                channel.trySend(list[it])
+            }
+        }
+    }
+
+
+    @Benchmark
+    fun sendUnlimited() = runBlocking {
+        runSend(count, Channel.UNLIMITED)
+    }
+
+    @Benchmark
+    fun sendConflated() = runBlocking {
+        runSend(count, Channel.CONFLATED)
+    }
+
+    @Benchmark
+    fun sendReceiveUnlimited(wrapper: UnlimitedChannelWrapper) = runBlocking {
+        runSendReceive(wrapper.channel, count)
+    }
+
+    @Benchmark
+    fun sendReceiveConflated() = runBlocking(Dispatchers.Default) {
+        runSendReceive(Channel(Channel.CONFLATED), count)
+    }
+
+    @Benchmark
+    fun sendReceiveRendezvous() = runBlocking(Dispatchers.Default) {
+        // NB: Rendezvous is partly benchmarking the scheduler, not the channel alone.
+        // So don't trust the Rendezvous results too much.
+        runSendReceive(Channel(Channel.RENDEZVOUS), count)
+    }
+
+    @Benchmark
+    fun oneSenderManyReceivers(wrapper: UnlimitedChannelWrapper) = runBlocking {
+        runSendReceive(wrapper.channel, count, 1, cores - 1)
+    }
+
+    @Benchmark
+    fun manySendersOneReceiver(wrapper: UnlimitedChannelWrapper) = runBlocking {
+        runSendReceive(wrapper.channel, count, cores - 1, 1)
+    }
+
+    @Benchmark
+    fun manySendersManyReceivers(wrapper: UnlimitedChannelWrapper) = runBlocking {
+        runSendReceive(wrapper.channel, count, cores / 2, cores / 2)
+    }
+
+    private suspend fun sendManyItems(count: Int, channel: Channel<Int>) {
+        repeat(count) {
+            // NB: it is `send`, not `trySend`, on purpose, since we are testing the `send` performance here.
+            channel.send(list[it])
+        }
+    }
+
+    private suspend fun runSend(count: Int, capacity: Int) {
+        Channel<Int>(capacity).also {
+            sendManyItems(count, it)
+        }
+    }
+
+    suspend fun <E> Channel<E>.forEach(action: (E) -> Unit) {
+        for (element in this) {
+            action(element)
+        }
+    }
+
+    // NB: not all parameter combinations make sense in general.
+    // E.g., for the rendezvous channel, senders should be equal to receivers.
+    // If they are non-equal, it's a special case of performance under contention.
+    private suspend inline fun runSendReceive(channel: Channel<Int>, count: Int, senders: Int = 1, receivers: Int = 1) {
+        require(senders > 0 && receivers > 0)
+        // Can be used with more than num cores but needs thinking it through,
+        // e.g., what would it measure?
+        require(senders + receivers <= cores)
+        // if the channel is prefilled, do not receive the prefilled items
```

Comment:
The prefilled items will be received, as the channel is FIFO, so the way I'd explain the logic I see here is that we only want to receive as many items as there were sent, which in case of a non-prefilled channel means, all the items.